### DISCUSSION TITLE
Nested x-data

### DISCUSSION THREAD

**Discussion by TheSteveBurgess** [alpinejs/alpine#3973] [1. Help]
With nested x-data is there a "correct" way to access both the child and parent version of a variable?

Take the below example

```
<div x-data="{
	variable: 'parent variable'
}">
	<div x-data="{
		variable: 'child variable',
		init(){
		  console.log(variable) # prints 'parent variable'
		  console.log(this.variable) # prints 'child variable'
		}
	}"></div>
</div>
```

Sometimes its useful to be able to see what the parent variable is even if there is a child variable with the same name. 

Is this approach OK to do? I don't want to start relying on it only to find it's a quirk of some implementation detail and gets patched out in a later version :)

**Comment by ekwoka** [comment]
You would not use the same names.

This is shadowing, and the expectation is that the local state is what you access.

**Reply by SteveBurgess** [reply]
Ok thanks.
